pkg/providers/amifamily/ami.go

```diff
@@ -72,7 +72,10 @@ func (p *DefaultProvider) List(ctx context.Context, nodeClass *v1.EC2NodeClass)
 	// Discover deprecated AMIs if automatic AMI discovery and upgrade is enabled. This ensures we'll be able to
 	// provision in the event of an EKS optimized AMI being deprecated.
 	includeDeprecated := lo.ContainsBy(nodeClass.Spec.AMISelectorTerms, func(term v1.AMISelectorTerm) bool {
-		return term.Alias == "latest"
+		if term.Alias == "" {
```

Comment:
Does passing an empty string to v1.AMIVersionFromAlias produce `latest`?

Reply:
No, this function has an invariant that it should be a non-empty string (and a valid alias). Right now this invariant is enforced by a panic.